libraries/rush-lib/src/logic/operations/OperationExecutionManager.ts

```diff
@@ -201,6 +220,12 @@ export class OperationExecutionManager {
       }
     );
 
+    if (!this._quietMode) {
+      if (this._projectLogFooter && this._lastTask) {
+        this._terminal.writeStdoutLine(this._projectLogFooter.replace(/\$\{taskName\}/g, this._lastTask));
```

Comment:
Seems like there should be a matching header here.

Reply:
In the "stream started" callback for each operation, we write the footer for last section, the separator, and then the header for the next section, and wait until all operations have executed the stream started callback.

At this point, all the operations are done, but we're still "inside" the last folding group that was started, so we need one final line to close it out (if defined).

Reply:
Does this mean that we don't fold the first operation that runs?

It seems like we should write the header, log stuff for the operation, and then write the footer (and it should appear that way in the code). Shouldn't stream-collator be able to handle that?

Reply:
Well, StreamCollator _is_ handling it -- it provides a callback called `onWriterActive` that triggers whenever a new stream starts writing to the collator (including the first stream). We pass this function as that callback, then we just pile all the build log streams into the collator, letting its `onWriterActive` hook write out the fancy header lines as appropriate.

The first stream to write to the collator will trigger this hook, `_lastTask` will be empty so it will skip writing a folding footer, then it will write the header separator, then it will write the folding header.